common/src/main/scala/uk/ac/wellcome/finatra/modules/DynamoConfigModule.scala

```diff
@@ -3,20 +3,53 @@ package uk.ac.wellcome.finatra.modules
 import javax.inject.Singleton
 
 import com.google.inject.Provides
+import com.twitter.app.Flag
 import com.twitter.inject.TwitterModule
 import uk.ac.wellcome.models.aws.DynamoConfig
 
-object DynamoConfigModule extends TwitterModule {
-  private val applicationName = flag[String]("aws.dynamo.streams.appName",
-                                             "dynamodb-streams-app",
-                                             "Name of the Kinesis app")
-  private val arn =
-    flag[String]("aws.dynamo.streams.arn", "", "ARN of the DynamoDB stream")
-  private val table =
-    flag[String]("aws.dynamo.tableName", "", "Name of the DynamoDB table")
+
+trait DynamoConfigModule extends TwitterModule {
+  val tableName: String
+
+  val applicationName: Flag[String]
+  val arn: Flag[String]
+  val table: Flag[String]
 
   @Singleton
   @Provides
   def providesDynamoConfig(): DynamoConfig =
     DynamoConfig(applicationName(), arn(), table())
+
+  val applicationNameTpl:String  = "aws.dynamo.%s.streams.appName"
+  val arnTpl: String = "aws.dynamo.%s.streams.arn"
+  val tableTpl: String = "aws.dynamo.%s.tableName"
+
+  def appNameFlagId = applicationNameTpl.format(tableName)
+  def arnFlagId = arnTpl.format(tableName)
+  def tableFlagId = tableTpl.format(tableName)
+}
+
+object MiroTableDynamoConfigModule extends DynamoConfigModule {
+  val tableName = "miroData"
+
+  val applicationName = flag[String](appNameFlagId, "", "Name of the Kinesis app")
```

Comment:
I tried various ways of pushing this functionality down into the trait but to no avail.